docs(www): Add hooks to static-query docs (#11974)
gatsbyjs/gatsby: docs/docs/static-query.md

Using `StaticQuery`, you can colocate a component with its data. No longer is it required to, say, pass data down from `Layout` to `Header`.

> ### Hooks Example

Reviewer: ```suggestion
### useStaticQuery
```

> Using the hooks version of static query is very similar. Checkout the documentation of `useStaticQuery` [here!](/docs/use-static-query/)

Reviewer: ```suggestion
There's also a hooks version of StaticQuery called `useStaticQuery`. Check out its documentation [here!](/docs/use-static-query/)
```
